pkg/cluster/cluster.go

```diff
@@ -390,6 +396,132 @@ func (c *Cluster) createPod(members etcdutil.MemberSet, m *etcdutil.Member, stat
+	netCases := []networkingv1.NetworkPolicy{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "sensu-block-all",
+				Namespace: metav1.NamespaceDefault,
+			},
+			Spec: networkingv1.NetworkPolicySpec{
+				PodSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app": "sensu",
+					},
+				},
+				Ingress: []networkingv1.NetworkPolicyIngressRule{},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "sensu-api-pods",
+				Namespace: metav1.NamespaceDefault,
+			},
+			Spec: networkingv1.NetworkPolicySpec{
+				PodSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app": "sensu",
+					},
+				},
+				Ingress: []networkingv1.NetworkPolicyIngressRule{
+					{
+						Ports: []networkingv1.NetworkPolicyPort{
+							{
+								Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 3000},
+							},
+							{
+								Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 8080},
+							},
+							{
+								Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 8081},
+							},
+						},
+						From: []networkingv1.NetworkPolicyPeer{},
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "sensu-operator-pods",
+				Namespace: metav1.NamespaceDefault,
+			},
+			Spec: networkingv1.NetworkPolicySpec{
+				PodSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app": "sensu",
+					},
+				},
+				Ingress: []networkingv1.NetworkPolicyIngressRule{
+					{
+						Ports: []networkingv1.NetworkPolicyPort{
+							{
+								Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 2379},
+							},
+							{
+								Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 2380},
+							},
+						},
+						From: []networkingv1.NetworkPolicyPeer{
+							{
+								PodSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										"name": "sensu-operator",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "sensu-cluster-pods",
+				Namespace: metav1.NamespaceDefault,
+			},
+			Spec: networkingv1.NetworkPolicySpec{
+				PodSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app": "sensu",
+					},
+				},
+				Ingress: []networkingv1.NetworkPolicyIngressRule{
+					{
+						Ports: []networkingv1.NetworkPolicyPort{
+							{
+								Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 2379},
+							},
+							{
+								Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 2380},
+							},
+						},
+						From: []networkingv1.NetworkPolicyPeer{
+							{
+								PodSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										"app": "sensu",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	c.logger.Infof("Creating NetworkPolicy...")
+	for _, net := range netCases {
+		_, err := c.config.KubeCli.NetworkingV1().NetworkPolicies(c.cluster.Namespace).Create(&net)
```

Comment:
Without a check if the network policy was already added, this will result in a warning for all clusters after the first:

```
time="2018-08-27T10:51:35Z" level=warning msg="network policy failed: networkpolicies.networking.k8s.io \"sensu-block-all\" already exists" cluster-name=example-sensu-cluster-2 pkg=cluster
```

I'm OK with leaving this for later, but maybe add a TODO, like: `TODO: only create NetworkPolicy if it doesn't exist already or, alternatively, create a unique policy per cluster`.

Reply:
I think I will be able to implement this by using `NetworkPolicies(c.cluster.Namespace).Get()` to check if its already deployed... I will try this